src/server/common.cc

```diff
@@ -270,7 +270,9 @@ SearchStats& SearchStats::operator+=(const SearchStats& o) {
   ADD(used_memory);
   ADD(num_entries);
 
-  DCHECK(num_indices == 0 || num_indices == o.num_indices);
+  // Different shards could have inconsistent num_indices values during concurrent operations.
+  // This can happen on concurrent index creation.
+  // We use max to ensure that the total num_indices is the maximum of all shards.
```

Comment:
That's a heuristic, we could also be deleting indices which upon this case it would be min. But it's fine -- this is the correct approach :)